### ISSUE TITLE
Add max heap to node environment data source

### ISSUE THREAD

**Issue by tunniclm** [RuntimeTools/appmetrics#53]
Provide the V8 max heap value in the node environment plugin data source. Since the max heap size will not change it should be fine to provide it once in the node environment data.

**Comment by tunniclm** [comment]
It is simple enough to expose the heap_size_limit() value that can be obtained from the V8 HeapStatistics API. However, investigation shows that this value is (4*max_semi_space_size + max_old_generation_size) which doesn't actually represent the maximum size of the heaps.

There is one old generation space and the young generation is made up of 2 semi spaces. The heap_size_limit() gives the amount of address space **reserved** for these, which includes extra alignment address space for the semi spaces -- V8 reserves double the semi space size for this purpose.

I think what we actually need to properly represent the max heap size is (2*max_semi_space_size + max_old_generation_size). I cannot find a simple way to calculate this, since neither the max_semi_space_size or max_old_generation_size do not seem to be exposed by the V8 public API.

**Comment by tunniclm** [comment]
Also, it would also be interesting to know the max_old_generation_size as a value independent of the calculation of the max heap size.

Unfortunately, the new HeapSpaceStatistics API does not provide maximum sizes for the spaces either.

**Comment by tunniclm** [comment]
It is possible to calculate the values for max old space and max semi spaces sizes based on replicating the behaviour of V8. I have raised a pull request with an implementation that appears to work for the versions of V8 current for Node v0.10, v0.12, v4.

It adds the following keys to the environment object/message:
```
heap.size.limit=<bytes>
max.old.space.size=<bytes>
max.semi.space.size=<bytes>
```
`heap.size.limit` is the heap_size_limit value from the HeapStatistics API (in bytes)
`max.old.space.size` is the calculated value of the old generation max size (in bytes)
`max.semi.space.size` is the calculated value of **half** the young generation max size (in bytes)

**Comment by seabaylea** [comment]
Resolved by PR #90

**Comment by seabaylea** [comment]
Note that we also added the following during the review cycle:
`max.heap.size=<bytes>`
which represents the maximum heap size value that the GC used value can be compared to.
